src/psygnal/containers/_evented_dict.py

```diff
@@ -76,19 +77,20 @@ def _type_check(self, value: _V) -> _V:
             )
         return value
 
-    def __newlike__(
-        self, mapping: MutableMapping[_K, _V]
-    ) -> "TypedMutableMapping[_K, _V]":
+    def __newlike__(self: _Cls, mapping: MutableMapping[_K, _V]) -> _Cls:
         new = self.__class__()
         # separating this allows subclasses to omit these from their `__init__`
         new._basetypes = self._basetypes
         new.update(mapping)
         return new
 
-    def copy(self) -> "TypedMutableMapping[_K, _V]":
+    def copy(self: _Cls) -> _Cls:
```

Comment:
not necessary if you don't want to but i think this is where `typing.Self` is useful?
https://docs.python.org/3/library/typing.html#typing.Self

Reply:
`typing.Self` is available since python 3.11. I know that it is available from `typing_extensions`. But I remember some problems. Especially when psygnal support python 3.7

Reply:
it would go in the `if TYPE_CHECKING` section.  psygnal supports 3.7 at runtime, but not for type checking.  We  use *many* recent typing constructs (including `ParamSpec`) that are only supported in later versions of python, so as long as you restrict your import to `TYPE_CHECKING` it's fine

Reply:
Changed